src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -779,6 +779,7 @@ public <T extends TransportRequest> TransportRequestHandler<T> interceptHandler(
 
                         @Override
                         public void messageReceived(T request, TransportChannel channel, Task task) throws Exception {
+                            threadContext.putTransient(ConfigConstants.USE_JDK_SERIALIZATION, channel.getVersion().before(Version.V_3_0_0));
```

Comment:
Please add a test to confirm this works as expected so we know if it is broken during a backport to a 2.x branch.

Reply:
I've moved this to `SecuritySSLRequestHandler` for testability and added a basic UT for verifying that the transient header is set when a message is received.

Since this is tagged for release in `2.11.0` now, I've updated the version checks. `useJDKSerialization` will be evaluated as true if the remote node version is < `2.11.0`. With this, the PR can be backported as is to `2.x` and no further changes shall be required in `main` as well.

Reply:
I was looking into `SecurityInterceptorTests` to cover the JDK vs Custom serialization changes and noticed that the tests are implemented incorrectly.

* There are assertions configured within the `doAnswer()` mocking for `sender`. These mockings are being setup **after** the calls to `SecurityInterceptor::sendRequestDecorate()` and hence the assertions never execute. 
* `isSameNodeRequest = false` scenario too is incorrectly implemented. The `otherNode` object is used for both setting up the `connection2` and passing as `localNode` parameter to `sendRequestDecorate`.

I'm fixing these tests and on top of it also adding the scenario where the remote node is on a version that support JDK serialization to cover the changes that I've made in SecurityInterceptor.